src/DotLiquid/Util/ObjectExtensionMethods.cs

```diff
@@ -50,11 +52,16 @@ public static object Send(this object value, string member, object[] parameters
         /// <returns>True if the values are equal, false otherwise.</returns>
         public static bool BackCompatSafeTypeInsensitiveEqual(this object value, object otherValue)
         {
-            // NOTE(daviburg): Historically testing for equality cross integer and string boundaries resulted in not equal.
-            // This ensures we preserve the behavior.
-            if (value is string && !(otherValue is char) || otherValue is string && !(value is char))
+            if (value != null && otherValue != null)
             {
-                return false;
+                // NOTE(daviburg): Historically testing for equality cross integer and string boundaries resulted in not equal.
+                // This ensures we preserve the behavior.
+                var aType = value.GetType();
+                var bType = otherValue.GetType();
+                if (aType != bType && (_numericTypes.Contains(aType) != _numericTypes.Contains(bType)))

```

Comment:
This is a significant departure from the earlier backward compatibility scope. The issue you have raised appears to be specific to testing contains in arrays for strings. But the code you wrote is a generic broad reaching type comparison for anything but 4 numeric types (which are not even all the numeric types in .NET).

Please scope the fix to the issue reported or,
- expand both the reported issue 
- expand validation tests to additional types
- learn historical behavior through testing

Otherwise we may just be creating yet more bugs to fix in the future.

Reply:
I went back to pre-362 and conducted several tests.
1. This fails: int testing - `AssertEvaluatesTrue("array", "contains", "1.0");` This is because 1.0 is not int and IList.Contains must have equal types.
2. String testing - all my tests are successful (not so on current tip)
3- Char testing - not supported (have no preference to this, but added because your code referred to char types). 'A' evaluates to string and IList.Contains must have equal types.

So technically speaking entire premise of 362 breaks backward compatibility. HOWEVER I understand fixing point 1 is a FEATURE and is GOOD. My code continues in that vain by defining numeric types that are compatible instead of testing char vs string to determine if it's numeric (which broke point 2)

Reply:
Let's discuss this elsewhere. I posted on gitter but happy to take this conversation to whatever is convenient.

Reply:
Yes, thank you for the investigation and will look up the conversation on gitter.

Reply:
Suggesting one-liner fix here:
if (value.GetType() != otherValue.GetType() && (value is string && !(otherValue is char) || otherValue is string && !(value is char)))
Simpler and better perf as we don't need to iterate through static arrays of types.

Reply:
As we discussed on gitter, this one-liner has a slew of issues. In the effort to maintain consistency with pre-362 behavior, all tests added to this commit are also added to https://github.com/microalps/dotliquid/tree/pre-362-tests for comparison.